packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -4856,9 +4865,7 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
                             value: _value,
                             cursorColor: _cursorColor,
                             backgroundCursorColor: widget.backgroundCursorColor,
-                            showCursor: EditableText.debugDeterministicCursor
```

Comment:
This is a code smell and would require some api change. We should reuse _cursorVisibilityNotifier and prevent update to it in _onCursorColorTick if EditableText.debugDeterministicCursor is true